### PR TITLE
Convert CPU ops to new form

### PR DESCRIPTION
**Pull request by diyessi**
Remove obsolete sigmoid

### CODE REVIEW

File: src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.hpp [hunk 1]
```diff
@@ -34,14 +34,17 @@ namespace ngraph
         class BatchMatMulTranspose : public Op
         {
         public:
+            NGRAPH_API
+            static const std::string type_name;
```
Comment (nmostafa) [inline src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.hpp]: Any reason this is puablic if we have `description()` ?

File: src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.cpp [hunk 2]
```diff
@@ -25,11 +25,13 @@
 using namespace std;
 using namespace ngraph;
 
-op::BatchMatMulTranspose::BatchMatMulTranspose(const shared_ptr<Node>& arg0,
-                                               const shared_ptr<Node>& arg1,
+const std::string op::BatchMatMulTranspose::type_name{"BatchMatMulTranspose"};
+
+op::BatchMatMulTranspose::BatchMatMulTranspose(const Output<Node>& arg0,
```
Comment (nmostafa) [inline src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.cpp]: One minor thing, all tests now rely on implicit conversion between `node` and `Output<Node>`. Maybe we should add a test to use `Output<Node>` directly when constructing nodes, and perhaps use non-zero index ?

Comment (diyessi) [reply src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.cpp]: Once all node class constructors have been upgraded, we can tighten things up some.

File: src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.hpp (hunk 1 above)
Comment (rkimballn1) [reply src/ngraph/runtime/cpu/op/batch_mat_mul_transpose.hpp]: With it public you can write something like `BatchMatMulTrasnpose::type_name` instead of typing the error prone literal string `"BatchMatMulTrasnpose"`. It is just a safer way to do it.
